src/cfnlint/maintenance.py

```diff
@@ -21,19 +22,37 @@
 def update_resource_specs():
     """ Update Resource Specs """
 
-    for region, url in SPEC_REGIONS.items():
-        filename = os.path.join(os.path.dirname(cfnlint.__file__),
-                                'data/CloudSpecs/%s.json' % region)
-        LOGGER.debug('Downloading template %s into %s', url, filename)
-        spec = json.loads(get_url_content(url))
+    # Pool() uses cpu count if no number of processors is specified
+    # Pool() only implements the Context Manager protocl from Python3.3 onwards,
+    # so it will fail Python2.7 style linting
+    # pylint: disable=not-context-manager
+    with multiprocessing.Pool() as p:
+        p.starmap(update_resource_spec, SPEC_REGIONS.items())
 
-        # Patch the files
-        spec = patch_spec(spec, 'all')
-        spec = patch_spec(spec, region)
+def update_resource_spec(region, url):
+    """ Update a single resource spec """
+    filename = os.path.join(os.path.dirname(cfnlint.__file__), 'data/CloudSpecs/%s.json' % region)
 
-        with open(filename, 'w') as f:
-            json.dump(spec, f, indent=2, sort_keys=True, separators=(',', ': '))
+    multiprocessing_logger = multiprocessing.log_to_stderr()
```

Comment:
One thing to note is I couldn't get logging to work inside the parallelised processes. Everything inside the parallel process is only debug logging, however it's definitely something that could be improved upon.

Reply:
Yesterday night  [I tried out a similar implementation to yours, but using multiple threads](https://github.com/aws-cloudformation/cfn-python-lint/pull/1410), and I think logging works there.

Reply:
I saw logs so I'm resolving this conversation.